### ISSUE TITLE
Windows lose saturation value after minimizing and reopen

### ISSUE THREAD

**Issue by rafaelsgirao** [WillPower3309/swayfx#134]
### Please read the following before submitting:
- Proprietary graphics drivers, including nvidia, are not supported. Please use the open source equivalents, such as nouveau, if you would like to use Sway. :heavy_check_mark: 
- Please do NOT submit issues for information from the github wiki. The github wiki is community maintained and therefore may contain outdated information, scripts that don't work or obsolete workarounds. :heavy_check_mark: 
  If you fix a script or find outdated information, don't hesitate to adjust the wiki page.

### Please fill out the following:
- **Swayfx Version:**
  - `sway -v`
```
/nix/store/dbs4173dk421ag660w0mpis6k3y1m9qj-sway-unwrapped-1.8🔒 via ❄️  impure (shell) 
❯ ./bin/sway -v
sway version 0.2
```
- **Debug Log:**
  - Run `sway -d 2> ~/sway.log` from a TTY and upload it to a pastebin, such as gist.github.com.
  - This will record information about sway's activity. Please try to keep the reproduction as brief as possible and exit sway.
  - Attach the **full** file, do not truncate it. :heavy_check_mark: 
https://gist.github.com/rafaelsgirao/700ba2636c8b0f4679eaa5d23f5e44a9

- **Configuration File:**
  - Please try to produce with the default configuration.
  - If you cannot reproduce with the default configuration, please try to find the minimal configuration to reproduce.
  - Upload the config to a pastebin such as gist.github.com. :heavy_check_mark: 
 Took default configuration, added the line `for_window [class="discord"] saturation set 0` .

- **Description:**
  - The steps you took in plain English to reproduce the problem. :heavy_check_mark: 
Minimizing or closing a window, letting it run in the background, then reopening causes the set saturation value to be lost.
Steps to reproduce:
  - Built swayfx from master (`nix build github:WillPower3309/swayfx`)
  - Take default Sway configuration, add a line to apply saturation=0 to a window, e.g (I'll use discord as an example):
```for_window [class="discord"] saturation set 0```
  - Close or minimize the window, reopen, and notice the saturation value that was set is now lost.

I also noticed both programs I tested this in (Discord and Ferdium) are both running under XWayland, if that matters - I don't know of any minimizable program that runs on native Wayland.

**Comment by WillPower3309** [comment]
Thanks for raising the issue! I'll ping @ErikReider on this, who implemented saturation and is also taking a look at scratchpad behavior

**Comment by ErikReider** [comment]
Close should work which is strange

**Comment by WillPower3309** [comment]
To clarify, would this occur when sending a window to the scratchpad? We (as in sway upstream as well) don't yet support minimization

**Comment by ErikReider** [comment]
Can't reproduce either way. Could you provide your config file?

**Comment by rafaelsgirao** [comment]
> To clarify, would this occur when sending a window to the scratchpad? We (as in sway upstream as well) don't yet support minimization

I can confirm that windows retain saturation when sending and retrieving them to scratchpad.

> Can't reproduce either way. Could you provide your config file?
Of course. But I've reproduced this on the default config ( just adding the line `for_window [class="discord"] saturation set 0` at the end ).  [Sway config used](https://gist.githubusercontent.com/rafaelsgirao/b9a116ac518189589b2befc859fcbcb4/raw/1e6ad1a9e0d2870a2ab03e966a740b6cdd72d1e6/sway-config)

**Comment by rafaelsgirao** [comment]
Adding some more information: Windows that lost their saturation get it back after I issue `swaymsg reload` and mess around with said window a little.

Also, checking the debug log I sent above again, I noticed these two snippets:
```txt
00:00:03.846 [DEBUG] [sway/tree/view.c:503] Checking criteria [class="discord"]
00:00:03.846 [DEBUG] [sway/tree/view.c:508] for_window '[class="discord"]' matches view 0x1d6d130, cmd: 'saturation set 0'
00:00:03.846 [INFO] [sway/commands.c:272] Handling command 'saturation set 0'
```
Then, down below (presumably after I've closed and reopened Discord from the tray bar), this snippet:
```txt
00:00:06.189 [DEBUG] [sway/tree/view.c:503] Checking criteria [class="discord"]
00:00:06.189 [DEBUG] [sway/tree/view.c:505] Criteria already executed
```

I think that from swayfx's perspective, the window is still the same (which seems plausible), so it doesn't reapply the lost saturation rate.

**Comment by ErikReider** [comment]
Hmmm. Seems to be an issue when closing the window into it's appindicator (which discord does by default)

**Comment by ErikReider** [comment]
This also applies to other for_window rules/preferences. One example would be my Steam Friends window rule `for_window [class="^Steam$" title="^Friends List$"] floating enable` which only works the first time I open the window. So this is a upstream bug

**Comment by WillPower3309** [comment]
Ah good to know, thanks for the investigation Erik, @rafaelsgirao try raising an upstream issue please :)
